WalletWasabi/WabiSabi/Client/WabiSabiHttpApiClient.cs

```diff
@@ -97,21 +97,9 @@ private async Task<HttpResponseMessage> SendWithRetriesAsync(RemoteAction action
 
 				return response;
 			}
-			catch (HttpRequestException e)
-			{
-				exceptions.Add(e);
-			}
 			catch (Exception e)
 			{
-				if (exceptions.Any())
-				{
-					exceptions.Add(e);
-					throw new AggregateException(exceptions);
-				}
-				else
-				{
-					throw;
-				}
+				exceptions.Add(e);
```

Comment:
I think it is rethrown, but in an aggregate exception with the rest of the exceptions:

If our cancellation token gets cancelled then the end of the do while (`} while (!combinedToken.IsCancellationRequested);`) stops the cycle and `throw new AggregateException(exceptions);` is called.

Reply:
In fact I added the task delay into a similar try-catch because that'd throw a `TaskCanceledException` and we'd lose all the exceptions come before otherwise: https://github.com/zkSNACKs/WalletWasabi/pull/8542/commits/efe63be57767b2a4f7d42524d32c66acadd445ca

Reply:
I agree with @kiminuo here.

Also, catching all exceptions always looks like a good idea but it is always a bad idea. I would prefer handling `TorConnectionException` and let `NullReferenceException`, `StackOverflowEception` and so on to throw.

Reply:
I don't see `TorConnectionException` in the stacktrace. I failed to find out what exception to catch. Please point me exactly what exception to handle and I'll do that.

Reply:
You're right. However `connection = await ObtainFreeConnectionAsync(request, circuit, cancellationToken).ConfigureAwait(false);` isn't in the try-catch and that can throw `TorException`s

Reply:
And indeed there's this in the logs! `at WalletWasabi.Tor.Socks5.Pool.TorHttpPool.ObtainFreeConnectionAsync(HttpRequestMessage request, ICircuit circuit, CancellationToken token) in WalletWasabi\Tor\Socks5\Pool\TorHttpPool.cs:line 245`

Reply:
Which is a `TorConnectCommandFailedException`. I think we got it now?